xla/service/gpu/runtime/gpublas_lt_matmul.cc

```diff
@@ -62,6 +62,27 @@ XLA_RUNTIME_REGISTER_ENUM_ATTR_DECODING(se::gpu::BlasLt::Epilogue);
 
 namespace gpu {
 
+MatmulPlanVec::Result MatmulPlanVec::GetOrCreate(int device_id, 
+      CreateFunc create_plan) {
+
+  se::gpu::BlasLt::MatmulPlan *plan_raw = nullptr;
+  {
+    absl::MutexLock lock(mu_.get());
+    if(device_id >= plans_.size()) {
+      plans_.resize(device_id + 1);
+    } else {
+      plan_raw = plans_[device_id].get();
+    }
+  }
+  if(plan_raw == nullptr) {
+    VLOG(2) << "Creating new MatmulPlan for device ID: " << device_id;
+    TF_ASSIGN_OR_RETURN(auto plan, create_plan());
+    absl::MutexLock lock(mu_.get());
+    plan_raw = plan.get();
+    plans_[device_id] = std::move(plan);
+  }
+}
```

Comment:
Do we need a return statement here?  "return plan_raw;"?

Reply:
ah, I am sorry for that: I have compiled the code but did not run the tests.. will fix it